### ISSUE TITLE
Specify fonts on a per-project basis

### ISSUE THREAD

**Issue by ararslan** [JuliaDocs/Documenter.jl#461]
Personally I quite dislike the default monospace font, Ubuntu Mono. I have no issues with the rest of the Documenter style, so what I'd like to be able to do is override a particular default, in this case just the font. (Unless of course you're open to a different default. 😉) Is this currently possible? Is there a way to e.g. inject
```css
pre, code {
  font-family: 'Source Code Pro', 'Andale Mono WT', courier, monospace !important;
}
```
?

**Comment by mortenpi** [comment]
Hmm, "Source Code Pro" is a bit sharper, which I actually like. It should be at `0.9em` though in my opinion, otherwise it feels a bit too large relative to the surrounding text. We could consider changing the font.

As for overriding the font for a repository -- it's doable but should be discouraged, so that the Julia package ecosystem would have a consistent look and feel. That being said:
* You could roll your own `documenter.css` under `docs/src/assets` -- this would replace the one provided by the package, and might break in the future when there are changes to the HTML.
* So the better option would be to have a custom `.css` file and use the `assets` keyword. I think we do get the ordering right, so the CSS in user-provided files takes precedence. The folks at Invenia are making use of that, e.g. https://invenia.github.io/Dispatcher.jl/latest/

One thing to consider is that for custom fonts there should be a link to a CDN as well, since it's likely that users don't have the fonts installed. Ideally the CDN link should go into a `<link>` tag on each of the HTML pages.. and that could be doable via the `assets` keyword as well, although absolute URLs might be broken at the moment (if it is it then that should be fixed). Otherwise, an `@import` in the beginning of your custom CSS file should do the trick as well.

**Comment by ararslan** [comment]
If you are interested in changing the font, I think Source Code Pro is a very nice choice; it has excellent Unicode support and I've yet to come across any character it didn't render properly. That's not the case for e.g. Inconsolata, which doesn't include Greek last I checked. Source Code Pro is also the font of choice for the Rust documentation, I believe.

> should be discouraged, so that the Julia package ecosystem would have a consistent look and feel

Hm, I'm not sure I agree, but that's neither here nor there.

Thanks for the heads up regarding Invenia's customization, I'll take a look at that.

**Comment by ararslan** [comment]
Is it documented somewhere why Documenter deviates from the style of the main Julia website and the current stable, Read the Docs-based documentation? That uses a serif font (Georgia I think?) with Andalé Mono WT, whereas Documenter uses a sans-serif font and Ubuntu Mono.

**Comment by mortenpi** [comment]
The original style, including fonts, was hashed out in #171 and then we switched from Fantasque to Ubuntu in #333, because Fantasque wasn't available in a reliable CDN. Sans-serif is used because they are considered to be more readable on screens compared to serif fonts. The Sphinx theme was used a lot for inspiration, but we also didn't hesitate to make improvements to it. We made no attempt to stay consistent with julialang.org.

As for particular fonts: Andale is not an open font AFAICT, so I would avoid it on those grounds (not that it couldn't be used, but just to have a principle with which to narrow down our choices). Inconsolata doesn't speak Greek as you say. Fantasque doesn't have a CDN available.

This leaves Source Code Pro (and Ubuntu). Comparing the fonts again, I still think that the former would be a slight improvement (if we set `font-size: 0.95em`), so I'd be happy to make the change. Would you like to open a PR for that? Changes would be needed in
 - [`assets/html/documenter.css#L24`](https://github.com/JuliaDocs/Documenter.jl/blob/0d77e0abbf2a545eba176a6c4d6d04baf039559c/assets/html/documenter.css#L24)
 - [`src/Writers/HTMLWriter.jl#L69`](https://github.com/JuliaDocs/Documenter.jl/blob/0d77e0abbf2a545eba176a6c4d6d04baf039559c/src/Writers/HTMLWriter.jl#L69)

And maybe the LaTeX style should be made consistent as well:
 - [`assets/latex/documenter.sty#L5`](https://github.com/JuliaDocs/Documenter.jl/blob/0d77e0abbf2a545eba176a6c4d6d04baf039559c/assets/latex/documenter.sty#L5)

**Comment by ararslan** [comment]
Great, thanks for the references. Nice to see the history and careful consideration behind all of the design.

I'd be happy to open a PR to use Source Code Pro. Another choice may be [Roboto Mono](https://fonts.google.com/specimen/Roboto+Mono), which is open, but I'm not certain on whether a CDN is available.

On a completely unrelated note, just an FYI, I've submitted a PR to highlight.js to update the Julia syntax definitions for 0.6. So once that lands and a new version is tagged there, Documenter should probably bump the highlight version used.

**Comment by mortenpi** [comment]
> Another choice may be Roboto Mono

Yeah, that one could be even better, although I don't actually have a strong preference -- both look great. Roboto Mono is available on Google Fonts, so it would fine in that respect.

> On a completely unrelated note, just an FYI, I've submitted a PR to highlight.js to update the Julia syntax definitions for 0.6. So once that lands and a new version is tagged there, Documenter should probably bump the highlight version used.

Good to know, thanks! Again feel free to PR once that's released, the version is specified here [`assets/html/documenter.js#L13-L14`](https://github.com/JuliaDocs/Documenter.jl/blob/0d77e0abbf2a545eba176a6c4d6d04baf039559c/assets/html/documenter.js#L13-L14)

Since it came up, do you have any thoughts on #385? I.e. that highlight.js doesn't support `jldoctest` blocks at the moment.

A small side-note -- julialang.org doesn't actually specify a font for `<pre>` tags (other than `monospace`). Your system just happens to default to Andale :smile: But making julialang.org consistent with the docs could be an option as well: https://imgur.com/a/O4PTK :laughing:

**Comment by ararslan** [comment]
Looking more at it, I'm actually thinking I like Roboto Mono more for pairing with Lato. If we were to go with that, do you think it should be similarly scaled to 0.95?

I'll look more at highlight.js. I wonder if they have an easy way to add a definition that's an alias of another. If so, that could be an easy fix, but if not then I'm not sure what the best course of action would be, since maintaining two separate definitions for the same language in highlight.js could be problematic.

I actually really like the julialang.org site like that! 😄

**Comment by mortenpi** [comment]
> Looking more at it, I'm actually thinking I like Roboto Mono more for pairing with Lato.

I'd be happy to go with that one as well.

> If we were to go with that, do you think it should be similarly scaled to 0.95?

Yep, I feel that otherwise inline code looks a too bit large compared to the surrounding Lato text.

**Comment by ararslan** [comment]
Okay, cool. I've submitted #463 to implement Roboto Mono, as a competitor to #462 (Source Code Pro). Let the battle begin! 😄

**Comment by mortenpi** [comment]
Fixed by #463.
